driver-core/src/main/java/com/datastax/driver/core/Responses.java

```diff
@@ -87,8 +87,8 @@ public DriverException asException(InetSocketAddress host) {
                 case PROTOCOL_ERROR:   return new DriverInternalError("An unexpected protocol error occurred. This is a bug in this library, please report: " + message);
                 case BAD_CREDENTIALS:  return new AuthenticationException(host, message);
                 case UNAVAILABLE:      return ((UnavailableException)infos).copy(); // We copy to have a nice stack trace
-                case OVERLOADED:       return new DriverInternalError(String.format("Queried host (%s) was overloaded; this shouldn't happen, another node should have been tried", host));
-                case IS_BOOTSTRAPPING: return new DriverInternalError(String.format("Queried host (%s) was bootstrapping; this shouldn't happen, another node should have been tried", host));
+                case OVERLOADED:       return new OverloadedException(host, message);
```

Comment:
So, due to JAVA-698 and after discussion with Olivier, I had to create a few new subclasses of DriverInternalError, in order to be able to distinguish between different types of errors that are reported to latency trackers. I created in total: OverloadedException, BootstrappingException and UnpreparedException.
One thing to notice is that for backwards compatibility reasons, they had to be subclasses of DriverInternalError.
We usually tend to consider DriverInternalError as a bug when it bubbles up into client code; but one must now admit that, since these errors are passed on to latency trackers, they can appear now in these trackers, even if the driver handled the error correctly (e.g. by retrying the query).